clusters/build-clusters/01_cluster/loki.yaml

```diff
@@ -0,0 +1,349 @@
+      app.kubernetes.io/part-of: loki
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: log-collector
+        app.kubernetes.io/instance: loki-promtail
+        app.kubernetes.io/name: promtail
+        app.kubernetes.io/part-of: loki
+        app.kubernetes.io/version: "2.2.1"
+    spec:
+      containers:
+      - args:
+        - --oidc.client-id=$(CLIENT_ID)
+        - --oidc.client-secret=$(CLIENT_SECRET)
+        - --oidc.issuer-url=https://sso.redhat.com/auth/realms/redhat-external
+        - --margin=10m
+        - --file=/tmp/shared/prod_bearer_token
+        name: prod-bearer-token
+        env:
+          - name: CLIENT_ID
+            valueFrom:
+              secretKeyRef:
+                name: promtail-prod-creds
+                key: client-id
+          - name: CLIENT_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: promtail-prod-creds
+                key: client-secret
+        volumeMounts:
+        - mountPath: "/tmp/shared"
+          name: shared-data
+        image: quay.io/observatorium/token-refresher
+      - command:
+        - sh
+        - -c
+        - |
+          promtail \
+            -client.external-labels=_id=$(cat /tmp/shared/cluster-id),host=$(HOSTNAME),cluster_name=$(CLUSTER_NAME) \
+            -config.file=/etc/promtail/promtail.yaml
+        env:
+        - name: HOSTNAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: CLUSTER_NAME
+          valueFrom:
+            configMapKeyRef:
+              key: cluster-name
+              name: cluster-name
+        image: quay.io/vrutkovs/promtail:2.2.1
+        imagePullPolicy: IfNotPresent
+        name: promtail
+        ports:
+        - containerPort: 3101
+          name: http-metrics
+        readinessProbe:
+          failureThreshold: 5
+          httpGet:
+            path: "/ready"
+            port: http-metrics
+          initialDelaySeconds: 10
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+        securityContext:
+          privileged: true
+          readOnlyRootFilesystem: true
+          runAsGroup: 0
+          runAsUser: 0
+        volumeMounts:
+        - mountPath: "/etc/promtail"
+          name: config
+        - mountPath: "/run/promtail"
+          name: run
+        - mountPath: "/var/lib/docker/containers"
+          name: docker
+          readOnly: true
+        - mountPath: "/var/log/pods"
+          name: pods
+          readOnly: true
+        - mountPath: "/tmp/shared"
+          name: shared-data
+        - mountPath: "/var/log/journal"
+          name: journal
+          readOnly: true
+      initContainers:
+      - command:
+        - sh
+        - "-c"
+        - oc get clusterversion/version -o=jsonpath='{.spec.clusterID}' > /tmp/shared/cluster-id && oc get cm openshift-install -n openshift-config -o=jsonpath='{.data.invoker}' > /tmp/shared/cluster-invoker
```

Comment:
Does this installation require the `invoker`? It looks not used anywhere here.

Reply:
Right, initContainer is required for cluster ID only, it can be removed